src/librustc_middle/mir/interpret/queries.rs

```diff
@@ -75,12 +75,22 @@ impl<'tcx> TyCtxt<'tcx> {
         }
     }
 
-    /// Evaluate a static, returning the allocation of the initializer's memory.
-    pub fn const_eval_static(self, def_id: DefId) -> Result<&'tcx Allocation, ErrorHandled> {
-        trace!("const_eval_static: Need to compute {:?}", def_id);
+    /// Evaluate a static's initializer, returning the allocation of the initializer's memory.
+    pub fn eval_static_initializer(self, def_id: DefId) -> Result<&'tcx Allocation, ErrorHandled> {
+        trace!("eval_static_initializer: Need to compute {:?}", def_id);
         let instance = ty::Instance::mono(self, def_id);
```

Comment:
Would this be a good place for another assertion making sure this is really a static?